### PR TITLE
Deal with optimizer.differentiable in PyTorch 1.13.0

### PR DESCRIPTION
**Pull request by comaniac**
Details are in #801.

cc @sgugger

### CODE REVIEW

**Comment by comaniac** [comment]
closing #801

File: src/accelerate/accelerator.py [hunk 1]
```diff
@@ -647,7 +647,15 @@ def _prepare_fsdp(self, *args):
                         "FSDP Warning: When using FSDP, several parameter groups will be conflated into "
                         "a single one due to nested module wrapping and parameter flattening."
                     )
-                optimizer = obj.optimizer.__class__(model.parameters(), **obj.optimizer.defaults)
+                try:
+                    optimizer = obj.optimizer.__class__(model.parameters(), **obj.optimizer.defaults)
+                except TypeError:
+                    if "differentiable" in obj.optimizer.defaults:
+                        # https://github.com/huggingface/accelerate/issues/801
+                        del obj.optimizer.defaults["differentiable"]
```
Comment (sgugger) [inline src/accelerate/accelerator.py:655]: This will delete inplace, we should probably create a new dict here

**Review by sgugger — COMMENTED** [review]
Thanks for working on this!

File: src/accelerate/accelerator.py [hunk 2]
```diff
@@ -647,7 +647,15 @@ def _prepare_fsdp(self, *args):
...
+                        optimizer = obj.optimizer.__class__(model.parameters(), **obj.optimizer.defaults)
```
Comment (comaniac) [inline src/accelerate/accelerator.py:656]:
```suggestion
                        defaults = obj.optimizer.defaults.copy()
                        del defaults["differentiable"]
                        optimizer = obj.optimizer.__class__(model.parameters(), **defaults)
```

File: src/accelerate/accelerator.py (hunk 1 above)
Comment (comaniac) [reply src/accelerate/accelerator.py]: Good point. Done. PTAL.

File: src/accelerate/accelerator.py [hunk 3]
```diff
@@ -647,7 +647,16 @@ def _prepare_fsdp(self, *args):
...
                         "a single one due to nested module wrapping and parameter flattening."
                     )
-                optimizer = obj.optimizer.__class__(model.parameters(), **obj.optimizer.defaults)
+                try:
+                    optimizer = obj.optimizer.__class__(model.parameters(), **obj.optimizer.defaults)
+                except TypeError:
+                    if "differentiable" in obj.optimizer.defaults:
+                        # https://github.com/huggingface/accelerate/issues/801
+                        defaults = obj.optimizer.defaults.copy()
+                        del defaults["differentiable"]
```
Comment (sgugger) [inline src/accelerate/accelerator.py:656]:
```suggestion
                        defaults = {k: v for k, v in obj.optimizer.defaults.items() if k != "differentiable"}
```

**Review by sgugger — APPROVED** [review]
One last nit :-)

**Comment by pacman100** [comment]
Thank you for the quick fix! 🤗

**Pull request merged**